### PR TITLE
feat(art): ui art babe!

### LINKED ISSUE #62: Streak and Combo information + UI enhencement
Like in the game, should we display the current streak of consecutive group completions and the combos(if any)  after placing a block?

### PR DESCRIPTION
**Pull request by Xiaorui-Huang**
enhence UI 

### Motivation and context

resolves #62 

### Describe your changes

new __str__ for ui and improve athethetics

### Tests performed

Manual UI test

### Before submitting this PR, please make sure:
 - [x] Your code builds cleanly without any errors

### CODE REVIEW

**Comment by Xiaorui-Huang** [comment]
image
image
image
image

**Comment by Xiaorui-Huang** [comment]
Game Over will be a random font to spice things up

**Comment by Xiaorui-Huang** [comment]
you can force quit using ctrl-c

File: woodoku/entity/woodoku_board.py
Comment (YANLING-H) [inline woodoku/entity/woodoku_board.py:193]: If you want to use macro BOARD_SIZE instead of 9, you need to change the in_between() function in util.py. Otherwise, by changing BOARD_SIZE, our board is messed up.
Screen Shot 2022-07-31 at 7 33 06 PM


Also, you may consider changing the padding between x coordinates and the game board, since for x > 9, the horizontal bars can't match those above. 

Screen Shot 2022-07-31 at 7 32 35 PM

File: woodoku/ui/command_line_ui.py
Comment (YANLING-H) [inline woodoku/ui/command_line_ui.py:68]: You might consider adding red crosses before any warning or removing this red cross, otherwise, the format of warning messages is not consistent

**Review by YANLING-H — CHANGES_REQUESTED** [review]

**Comment by YANLING-H** [comment]
I feel a bit distracting when seeing a large art of "score: 0" and "game over". @EdwardHaoranLee  may share any idea about it.

File: woodoku/entity/woodoku_board.py
Comment (Xiaorui-Huang) [reply woodoku/entity/woodoku_board.py:193]: No, the macro are not meant to be changed, Macros are constants, they will stay with a value of 9. The use here is simply to convey the meaning that 9 is the border size. 

and our game simply don't support any thing other than 9. That is not the game

Comment (YANLING-H) [reply woodoku/entity/woodoku_board.py:193]: Ok, it makes sense

**Comment by Xiaorui-Huang** [comment]
> I feel a bit distracting when seeing a large art of "score: 0" and "game over". @EdwardHaoranLee may share any idea about it.

As in distracted by what? Games over... you are meant to see what final score you got and that the game is over

File: woodoku/ui/command_line_ui.py
Comment (Xiaorui-Huang) [reply woodoku/ui/command_line_ui.py:68]: My reasoning is that the  "cannot place shape error" error is too small compared to the arts. So the cross is there give it more attention. The other errors are directly after when you input something wrong, there is no need for extra attention and bloating the error messages. And the reason I didn't make the "cannot place shape error"  use art is that the the error is too long and arts have way to large of a font

Comment (YANLING-H) [reply woodoku/ui/command_line_ui.py:68]: Ok, it makes sense

**Comment by YANLING-H** [comment]
> > I feel a bit distracting when seeing a large art of "score: 0" and "game over". @EdwardHaoranLee may share any idea about it.
> 
> As in distracted by what? Games over... you are meant to see what final score you got and that the game is over

The size of the art makes sense when running the game on a larger screen, so there is no problem anymore.

**Review by YANLING-H — DISMISSED** [review]

**Comment by Xiaorui-Huang** [comment]
Important design change. Will also wait for @EdwardHaoranLee 's approval

File: woodoku/entity/score_agent.py
Comment (EdwardHaoranLee) [inline woodoku/entity/score_agent.py]: Why calling method instead of accessing the instance variables?

**Review by EdwardHaoranLee — DISMISSED** [review]

File: woodoku/entity/score_agent.py
Comment (Xiaorui-Huang) [reply woodoku/entity/score_agent.py]: This is for mocking purposes. Since get is public methods, I can use them in combination with calculating winning to more easily set the private state of a score agent. Otherwise, it would be the current implementation that reserves engineer the desired score and how to set the state only with calculate winning(which can't be mocked since it's a void method)

Comment (Xiaorui-Huang) [reply woodoku/entity/score_agent.py]: I'd need these get method to implement the mocking in #61 so I'd probably will just copy/paste these method or cherry pick the commit of these changes to #61methods

Comment (Xiaorui-Huang) [reply woodoku/entity/score_agent.py]: reversing course on mocking see comment[https://github.com/EdwardHaoranLee/WoodokuLearn/pull/61#discussion_r936237242](url) here, so I will reverse and access instance variables directly again

**Review by EdwardHaoranLee — APPROVED** [review]

**Pull request merged**
